src/groups.ts

```diff
@@ -55,25 +55,29 @@ export const selectGroup = (
   };
 };
 
-export const selectGroups = (function () {
+export const selectGroupsForSelectedElements = (function () {
+  type SelectGroupsReturnType = Pick<
+    InteractiveCanvasAppState,
+    "selectedGroupIds" | "editingGroupId" | "selectedElementIds"
+  >;
+
   let lastSelectedElements: readonly NonDeleted<ExcalidrawElement>[] | null =
     null;
   let lastElements: readonly NonDeleted<ExcalidrawElement>[] | null = null;
-  let lastAppState: InteractiveCanvasAppState | null = null;
+  let lastReturnValue: SelectGroupsReturnType | null = null;
 
-  const ret = (
+  const _selectGroups = (
```

Comment:
moved the `selectGroups` helper inside the private scope since it was previously exported but wasn't meant to be used publicly